draft-trammell-taps-post-sockets.md

```diff
@@ -818,12 +829,133 @@ func sayHelloWithAssociation(association Association) {
 }
 ~~~~~~~~
 
+## API Dynamics {#sec-dynamics}
+
+As Carriers provide the central entry point to Post, they are key to API
+dynamics. The lifecycle of a carrier is shown in {{fig-carrier-lc}}. Carriers
+are created by active openers by calling Initiate() given a Local and a Remote,
+and by passive openers by calling Listen() given a Local; the .Accept() method
+on the listener Carrier can then be used to create active carriers. By default,
+the underlying Association is automatically created and managed by the
+underlying API. This underlying Association can be accessed by the Carrier's
+.Association() method. Alternately, an association can be explicitly created
+using NewAssociation(), and a Carrier on the association may be accessed or
+initiated by calling the association's .Initiate() method.
+
+Once a Carrier has been created (via Initiate(), Association.Initiate(),
+NewSource(), NewSink(), or Listen()/Accept()), it may be used to send and
```

Comment:
The naming seems inconsistent. Should NewSource() and NewSink() be InitiateAsSource() and InitiateAsSink()?

Reply:
Eh... you're not initiating anything as a Sink. Maybe as a source, but not always (i.e. preparing for multicasting). So I think these are more correct that InitiateAs..., even if still clunky.

Reply:
I'd leave as New for now. Initiate is not the right verb; Create/Instantiate/Prepare maybe, but New is fine.